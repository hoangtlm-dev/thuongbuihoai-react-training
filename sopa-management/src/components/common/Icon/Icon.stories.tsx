import type { Meta, StoryObj } from '@storybook/react';
import Icon, { IconType } from '.';

const meta: Meta<typeof Icon> = {
  component: Icon,
};

export default meta;
type Story = StoryObj<typeof Icon>;

export const IconCart: Story = {
  args: {
    iconName: IconType.user
  }
};
