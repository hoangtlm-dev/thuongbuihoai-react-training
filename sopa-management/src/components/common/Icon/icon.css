.icon-user {
  background-image: url(/images/icons/user.svg);
  display: inline-block;
  width: var(--w-small);
  height: var(--h-small);
}

.icon-trash {
  background-image: url(/images/icons/trash.svg);
  display: inline-block;
  width: var(--w-tiny);
  height: var(--h-tiny);
}

.icon-cart {
  background-image: url(/images/icons/shopping-bag.svg);
  display: inline-block;
  width: var(--w-small);
  height: var(--h-small);
}
