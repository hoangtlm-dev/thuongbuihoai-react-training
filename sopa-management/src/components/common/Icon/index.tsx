import React, { memo, MouseEventHandler } from "react";
import "./icon.css";

export enum IconType {
  user = "user",
  cart = "cart",
  trash = "trash",
};

interface IconProps {
  iconName?: IconType;
  onClick?: MouseEventHandler<HTMLElement>;
};

const Icon: React.FC<IconProps> = ({
  iconName,
  onClick = () => {},
}) => {
  let className = "icon";
  switch (iconName) {
    case IconType.user:
      className += ` icon-${IconType.user}`;
      break;
    case IconType.cart:
      className += ` icon-${IconType.cart}`;
      break;
    case IconType.trash:
      className += ` icon-${IconType.trash}`;
      break;
    default:
      break;
  };

  return (
    <div
      data-testid="icon"
      className={`${className} icon-${iconName}`}
      onClick={onClick}
    />
  )
};

export default memo(Icon);
