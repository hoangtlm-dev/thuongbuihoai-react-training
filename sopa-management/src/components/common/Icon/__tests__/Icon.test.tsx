import { fireEvent, render } from "@testing-library/react";
import "@testing-library/jest-dom";
import Icon, { IconType } from "../index";

describe("Icon component", () => {
  test("Should render Icon component with cart", () => {
    const { getByTestId } = render(<Icon iconName={IconType.cart} />);
    const icon = getByTestId("icon");
    fireEvent.click(icon);
    expect(icon).toBeInTheDocument();
  });

  test("Should render Icon component with trash", () => {
    const { getByTestId } = render(<Icon iconName={IconType.trash} />);
    const icon = getByTestId("icon");
    fireEvent.click(icon);
    expect(icon).toBeInTheDocument();
  });

  test("Should render Icon component with user", () => {
    const { getByTestId } = render(<Icon iconName={IconType.user} />);
    const icon = getByTestId("icon");
    fireEvent.click(icon);
    expect(icon).toBeInTheDocument();
  });

  test("Should render Icon component with default", () => {
    const { getByTestId } = render(<Icon />);
    const icon = getByTestId("icon");
    fireEvent.click(icon);
    expect(icon).toBeInTheDocument();
  });

  test("Matches snapshot", () => {
    const { container } = render(<Icon iconName={IconType.cart} />);
    expect(container).toMatchSnapshot();
  });
});
