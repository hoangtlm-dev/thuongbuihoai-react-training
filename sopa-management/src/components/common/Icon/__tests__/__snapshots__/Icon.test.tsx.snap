// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Icon component Matches snapshot 1`] = `
<div>
  <div
    class="icon icon-cart icon-cart"
    data-testid="icon"
  />
</div>
`;
